lib/SparqlGenerator.js

```diff
@@ -31,11 +33,12 @@ Generator.prototype.toQuery = function (q) {
   if (q.distinct)
     query += 'DISTINCT ';
 
-  if (q.variables)
+  if (q.variables){
     query += mapJoin(q.variables, undefined, function (variable) {
-      return isString(variable) ? this.toEntity(variable) :
-             '(' + this.toExpression(variable.expression) + ' AS ' + variable.variable + ')';
+      return isString(variable) || isTerm(variable) ? this.toEntity(variable) :
```

Comment:
@RubenVerborgh Do you want everything to be backwards-compatible with the string representation? Or can everything become terms?

@simonvbrae If so, you can remove all the `isString` stuff, and assume variables are always terms.

Reply:
Well, there is the case where variables are used as keys to an object, but I don't know if that's what's happening here.

Reply:
I have replaced the check with `=== '*'` so the check is a bit more strict but that's the best we can do since `'*'` is a valid variable at the moment.